### DISCUSSION TITLE
Getting deviceId and localKey directly with API calls

### DISCUSSION THREAD

**Discussion by SleepWalker** [codetheweb/tuyapi#435] [General]
Hello, I've found out, that an official Smart Life app shows you `deviceId`. I've tested that `deviceId` with tuya OpenAPI and it works (at least for my smart sockets). This means, that you can get `localKey` from an OpenAPI `/v1.0/devices/{{device_id}}`.

I think, that this may be useful to simplify `tuyapi` setup workflow.

The new setup will consist of three steps:

- after creating app on iot.tuya.com and getting key/secret, you need to link your devices, but instead of doing that with `@tuyapi/cli`, you should "Link device by Tuya Smart App Account":
![image](https://user-images.githubusercontent.com/2536916/82340075-7c52e280-99f7-11ea-95d6-672793cb7e79.png)
- then you can grab id from an app. From the main screen tap on device you interested in, then tap "edit icon" to get to the screen where you can manage/share/rename device. On that screen there is "Device information" menu option, where you can look up for the `deviceId`:
![image](https://user-images.githubusercontent.com/2536916/82341979-c341d780-99f9-11ea-8cb3-52062c40cfc6.png)
- finally request device info from tuya OpenAPI, where you can get `localKey` value:
![image](https://user-images.githubusercontent.com/2536916/82341488-28e19400-99f9-11ea-943e-2b080280b059.png)

**Comment by Jaja0624** [comment]
I tried this. I am getting: **There was an error in evaluating the Pre-request Script:  TypeError: Cannot read property 'sigBytes' of undefined.** when trying to make the get request. I also can't tell what I am suppose to use as schema value when making the request.

**Comment by codetheweb** [comment]
This could be really useful @SleepWalker, thank you for researching it.

`tuya-cli link` still has issues occasionally for some users, so this could be a great alternative.

I'll take a look at some point this weekend.

**Comment by etttoutoi** [comment]
Hello.

i would try your method, but im noob . It's ok for my virtual id, i have found.

But can you tell me how i have to find local key.

I don't know how i can use tuya openapi.

Can you tell me step by step what i have to write to found please.

thank you

**Comment by Moudoux** [comment]
@etttoutoi This [quick start guide](https://docs.tuya.com/en/iot/open-api/quick-start/quick-start1?id=K95ztz9u9t89n) from tuya has all the info you need for using postman to get the local key. Basically:

1. Install postman
2. Download and import the [environment](https://images.tuyacn.com/smart_res/developer_default/OpenAPI_postman_environment.zip) and [collection](https://images.tuyacn.com/smart_res/developer_default/OpenAPI_postman_collection.zip) in postman
3. Set up the variables according to your tuya developer profile `client_id, device_id (from the app as described in this post), schema, and url (region based, i.e. https://openapi.tuyaeu.com for EU)`
4. Run the `access_token` task in `collections` in postman
5. Run the `{{url}}/v1.0/devices/{{device_id}}` request to get your `local_key`

**Comment by SleepWalker** [comment]
Here is a gist, that demonstrates how to get `localKey` and then tries to use it with a demo code from `tuyapi` README:

https://gist.github.com/SleepWalker/18f19679adbfacee6c415d4f6dc218bb#file-readme-md

I've successfully turned on/off my socket using `demo.js` from that gist :tada:

**Comment by codetheweb** [comment]
@SleepWalker so after looking at this a bit, it seems that while linking your TuyaSmart account to your SaaS instance works fine, there's not a programmatic way to get all device IDs. `/devices` throws an error, and `/devices?schema=TuyaSmart` gives a permission error.  Did you find a good way to list devices?

Ideally, the user would be able to link their account through the web interface and then run one `tuya-cli` command to list devices; without having to manually find and copy device IDs.

**Comment by SleepWalker** [comment]
@codetheweb you can use `{{url}}/v1.0/users/{{uid}}/devices` for that. But, the only way to get `uid` I know, is by copying `deviceId` from an app and than get `uid` field from device info request.

So the user should copy `deviceId` from any random device, that he had connected to his account. And then it can be used to list all other devices

I've added demo script in gist, that lists all the available devices:

https://gist.github.com/SleepWalker/18f19679adbfacee6c415d4f6dc218bb#file-listdevices-js

**Comment by Moudoux** [comment]
deviceId can be obtained programmatically using the cloudtuya library, by providing it with your account login details (smart_life, tuya smart app) it will list the device ids for all connected devices, which can then be used here.

**Comment by tim-jc** [comment]
So close, but so far on this one. Fell at running `{url}}/v1.0/devices/{{device_id}}` with the error below. The `access_token` task had run successfully so I have my Postman environment set up correctly. Anybody seen this error?
![image](https://user-images.githubusercontent.com/65184272/84564154-aed2cf80-ad58-11ea-907d-6aec64a4e33e.png)

**Comment by SleepWalker** [comment]
@tim-jc looks like you've forgot to link your smart life account to your tuya dev account (https://auth.tuya.com/). See the issue description for the screenshots.

**Comment by aidbish** [comment]
What do i put for the schema. 
i am struggling to get token auth. i get 
{
    "code": 1000,
    "msg": "data not exist",
    "success": false,
    "t": 1592041129210
}

**Comment by tim-jc** [comment]
@SleepWalker - thank you. I had linked, but when I went back in to my Tuya account I saw my device count was now 0. I think I'd inadvertently run the `DEL` version of `{url}}/v1.0/devices/{{device_id}}` initially rather than `GET`. Re-added the device in the Tuya app and was able to get the key. Appreciate the help.

**Comment by aidbish** [comment]
@tim-jc 

what did u add for the schema to get auth token?

**Comment by tim-jc** [comment]
@aidbish - follow the instructions [here](https://github.com/codetheweb/tuyapi/blob/master/docs/SETUP.md). The Channel ID (step 3) is the value to use as the schema.

**Comment by aidbish** [comment]
hmm, tried that, still getting same error 100

i wonder if there is a delay before device being available, although it says 1 device
which is correct in the app

**Comment by codetheweb** [comment]
I plan to get to this next weekend, I've been busy with other stuff.

**Comment by cdprince** [comment]
I tried to use the gist.github link. I was not too sure about the .env file 
I was confused about: SOCKET_DEVICE_ID 
SOCKET_DEVICE_ID == <from smart-life application: "Virtual ID"> = 73328546dc4*******a0

Here is what I got:
pi@raspberrypi:~/git/18f19679adbfacee6c415d4f6dc218bb $ node listDevices
ERROR { Error: Error requesting device: permission deny (1106)
    at assertSuccess (/home/pi/git/18f19679adbfacee6c415d4f6dc218bb/tuyaApi.js:143:19)
    at TuyaApi.getDevice (/home/pi/git/18f19679adbfacee6c415d4f6dc218bb/tuyaApi.js:92:5)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  resp:
   { code: 1106,
     msg: 'permission deny',
     success: false,
     t: 1593044559653 } }
pi@raspberrypi:~/git/18f19679adbfacee6c415d4f6dc218bb $ node listDevices.js
ERROR { Error: Error requesting device: permission deny (1106)
    at assertSuccess (/home/pi/git/18f19679adbfacee6c415d4f6dc218bb/tuyaApi.js:143:19)
    at TuyaApi.getDevice (/home/pi/git/18f19679adbfacee6c415d4f6dc218bb/tuyaApi.js:92:5)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  resp:
   { code: 1106,
     msg: 'permission deny',
     success: false,
     t: 1593044623035 } }

...maybe I am doing something wrong.

Interesting that afterward I tried using the tuya-cli link and it worked on a device I could not get to work with the tuya-cli link formerly.

**Comment by SleepWalker** [comment]
@codetheweb in case it will be helpful, I've released my current API client implementation: https://github.com/SleepWalker/tuya-cloud-api

---

@cdprince `SOCKET_DEVICE_ID` is the same as `deviceId` from the first post of this issue. `permission deny` error is possibly because you haven't linked your tuya smart app account to tuya dev account (see first post)

**Comment by codetheweb** [comment]
I think there may be some overlap between that and 'my' [client](https://github.com/TuyaAPI/openapi), but nice. 😛 

Still planning to work on this tomorrow, hopefully will get it in the official setup instructions. 👍

**Comment by codetheweb** [comment]
As promised, I updated the OpenAPI and cli packages to support this use case. Also updated the [setup instructions](https://github.com/codetheweb/tuyapi/blob/master/docs/SETUP.md).

Please upgrade `tuya-cli` to >= v1.13.0 and give it a go. If you run into any problems, please open a new issue.

Thanks for the contributions @SleepWalker!

**Comment by Nodi-Rubrum** [comment]
Will someone PLEASE step-by-step illustrate how to this... "finally request device info from tuya OpenAPI, where you can get localKey value:
image"  You post a print screen with NO STEPS OR EXPLANATION AT ALL.  HELP!  You explained it well up to that step... and left a poor new user hanging with no clue how to use the OpenAPI.

**Comment by SleepWalker** [comment]
that text was written for the developers, who are familiar with existing tuya APIs and their official Postman collection. You can try to follow an official tuya docs about postman usage: https://developer.tuya.com/en/docs/iot/open-api/quick-start/quick-start1?id=K95ztz9u9t89n#title-4-Use%20Postman%20to%20simulate%20OpenApi%20to%20control%20the%20device

But, if you are planning to use `codetheweb/tuyapi`, you don't need that docs and any information listed in this issue. You should follow the following instructions: https://github.com/codetheweb/tuyapi/blob/master/docs/SETUP.md

**Comment by kinstray** [comment]
> @etttoutoi This [quick start guide](https://docs.tuya.com/en/iot/open-api/quick-start/quick-start1?id=K95ztz9u9t89n) from tuya has all the info you need for using postman to get the local key. Basically:
> 
> 1. Install postman
> 2. Download and import the [environment](https://images.tuyacn.com/smart_res/developer_default/OpenAPI_postman_environment.zip) and [collection](https://images.tuyacn.com/smart_res/developer_default/OpenAPI_postman_collection.zip) in postman
> 3. Set up the variables according to your tuya developer profile `client_id, device_id (from the app as described in this post), schema, and url (region based, i.e. https://openapi.tuyaeu.com for EU)`
> 4. Run the `access_token` task in `collections` in postman
> 5. Run the `{{url}}/v1.0/devices/{{device_id}}` request to get your `local_key`

Kudos for this!  This works great for getting your local_key without proxy's (Oct 2020).  Only thing to note is while in cloud development, on the Tuya site, make sure you go into 'API Group' and enable the permissions with a random reason.  Not overly sure which ones, but wait at least 5 to 10 minutes for the permissions to take effect then your 'postman' requests will work, giving you the local_key.

Well done that op!

**Comment by kinstray** [comment]
Rejigged the earlier comment with some extra stuff I had to do.  Thanks to @Moudoux for the info!

I found this method the easiest

Follow this guide first: https://developer.tuya.com/en/docs/iot/open-api/quick-start/quick-start1?id=K95ztz9u9t89n

Make sure you import your devices using 'Link devices by App account'.  Also under 'API Group', enable the API's giving any random reason.  I'm not sure which, I just enabled them all.  Remember there are 3 pages!

You'll need your Client ID, Client Secret and Device ID for the plug you need the Local ID for.  Those are all in the Tuya [portal](https://iot.tuya.com/cloud/).

<Mostly> Written by @Moudoux on 23 May, thanks to them for this bit! :

1. Install [Postman](https://www.postman.com/downloads/)
2. Download and import the [environment and collection](https://images.tuyacn.com/smart/docs/API_postman_collection.json) in postman
3. Set up the variables according to your [Tuya developer profile](https://iot.tuya.com/cloud/) client_id, device_id (from the app as described in this [post](https://developer.tuya.com/en/docs/iot/open-api/quick-start/quick-start1?id=K95ztz9u9t89n)) and url (region based, i.e. https://openapi.tuyaeu.com for EU) - Make sure you've enabled the API's (mentioned above) and wait 5 to 10 mins or so for them to take effect!
4. Run the access_token task in collections in postman
5. Run the {{url}}/v1.0/devices/{{device_id}} request to get your local_key

Found this much easier than messing around with SSL certificates on Android devices and Proxy apps!

**Comment by dsschneidermann** [comment]
For anyone interested and **having read the instructions in the previous post,** a full list of devices can be retrieved from the endpoint:
`{{url}}/v1.0/users/{{uid}}/devices`

Uid must be set to a user id from endpoint:
`{{url}}/v1.0/apps/{{schema}}/users?page_no=1&page_size=10`

**Comment by rpelletierrefocus** [comment]
I do not understand what I need to put in the easy_sign and t fields in the Headers. Any help would be much appreciated.

**Comment by Bill-Dung** [comment]
Hmmm, please help me, 

Get Access Token in Collection says 
   "success" : true 
but Get {{url}}/v1.0/devices/{{device_id}} says 
   "msg": "token invalid",
   "success": false,

What am I doing wrong?

**Comment by kontonaspam** [comment]
> 
> 
> I do not understand what I need to put in the easy_sign and t fields in the Headers. Any help would be much appreciated.

You need to include 't' probably because app on server side will check if it ever sent request with such timestamp. This is to limit flooding, DDOS, etc.

**Comment by kubalakowski** [comment]
> Hmmm, please help me,
> 
> Get Access Token in Collection says
> "success" : true
> but Get {{url}}/v1.0/devices/{{device_id}} says
> "msg": "token invalid",
> "success": false,
> 
> What am I doing wrong?

I'm using iot.tuya.com API Explorer and the same issue appears - "token invalid". Tried all of the available Data Centers, but with no success.
I have enabled all of the options under "Api Group"

EDIT: Although everything works fine when called via Postman

**Comment by kinstray** [comment]
Hi

It sounds like you need to run the 'Get access_token in Simple mode' task first.  This runs through the following code:

![2020-12-20 13_26_41-Window](https://user-images.githubusercontent.com/41172769/102714601-9e7f4c80-42c7-11eb-938b-f127c8c17609.png)

This will set the internal variable 'easy_access_token' and 'easy_refresh_token' for the use in future commands.  The token will also time out after a couple of hours too, so if you leave it and come back later, you need to re-run the 'Get access_token in Simple Mode' task again to get a new token.

You'll set you URL variable in the little 'eye' icon at the top right of Postman:

![image](https://user-images.githubusercontent.com/41172769/102714641-f0c06d80-42c7-11eb-850a-8747f5f60dd1.png)

This was step 4, here: https://developer.tuya.com/en/docs/iot/open-api/quick-start/quick-start1?id=K95ztz9u9t89n#title-6-Simulated%20call

Once you've went through all this pain, it pretty much runs itself!  My IP Address was changing on the device so you will have to read up on giving it a reservation in your router.  I can't post that here as every router will be different, sorry.  I can help you with a Virgin Media UK Hub v4?!?

Good luck, let me know how it goes!

**Comment by cinghialino** [comment]
> So close, but so far on this one. Fell at running `{url}}/v1.0/devices/{{device_id}}` with the error below. The `access_token` task had run successfully so I have my Postman environment set up correctly. Anybody seen this error?
> ![image](https://user-images.githubusercontent.com/65184272/84564154-aed2cf80-ad58-11ea-907d-6aec64a4e33e.png)

where is this screenshot from?

**Comment by Thunderbird2086** [comment]
It's from `postman`, a standalone application. 
> > ![image](https://user-images.githubusercontent.com/65184272/84564154-aed2cf80-ad58-11ea-907d-6aec64a4e33e.png)
> 
> where is this screenshot from?

**Comment by kinstray** [comment]
Permission Deny is caused because by default, access to all the Tuya Cloud API's is disabled.  You have to 'apply' for API Permission and give a reason.  This seems to be an automated process.  I wasn't sure which I needed, so I did them all (there are 3 pages worth!) They've changed the user interface slightly, they are here:

Inside the Tuya IoT Platform > Cloud > API Groups

![2021-01-05 08_32_22-Window](https://user-images.githubusercontent.com/41172769/103624459-76335700-4f31-11eb-8ad3-89ad0e840ff6.png)

It doesn't look like they've updated the documentation, but it's here: https://developer.tuya.com/en/docs/iot/open-api/quick-start/quick-start1?id=K95ztz9u9t89n#title-5-Unlock%20development%20capability

**Comment by DEVAXTATOR** [comment]
![Captura de ecrã 2021-01-20 190152](https://user-images.githubusercontent.com/59897162/105222175-05f01c80-5b52-11eb-8627-2bd2c5fb632f.png)
go here
and paste the device id
dont forge the zone

**Comment by poelman-git** [comment]
Hero!

**Comment by benov84** [comment]
Doesn't work any more :(

**Comment by DEVAXTATOR** [comment]
> Doesn't work any more :(

what? mine is working do as in the print screen.. use Europe and get device details remember to grant permissions to do that... they ate many guides to do this i made it work and i suck at programing and apis and crap like that... dont give up keep learning

**Comment by dwarf2k** [comment]
Re: [codetheweb/tuyapi] A new way of getting deviceId and localKey (#323)



3

Thank you,
Scott

On Wednesday, February 10, 2021, 2:53:04 PM, you wrote...




Doesn't work any more :(

**Comment by benov84** [comment]
> > Doesn't work any more :(
> 
> what? mine is working do as in the print screen.. use Europe and get device details remember to grant permissions to do that... they ate many guides to do this i made it work and i suck at programing and apis and crap like that... dont give up keep learning

Thank you! I made it work. My account was with chinese region, now with my europe acount, it works! :)

**Comment by Senbei123** [comment]
>     5\. {{url}}/v1.0/devices/{{device_id}}

I'm so happy to get this information from you!
All other methods reference to CLI tools - which I can't install!

Once again, thanks for this - after 2hrs of trying around this was the final and good hint!

I created a german video about the solution here: https://youtu.be/dBtrsXZOk8M

**Comment by 1aranzant** [comment]
A lot of users and I cannot use the full features of the Tuya IOT platform, the website gives the following error: 

![image](https://user-images.githubusercontent.com/17981079/108375796-9c236b00-7202-11eb-8bb3-bc7d6bdf6b46.png)

**Comment by DEVAXTATOR** [comment]
you need to link devices first
![Captura de ecrã 2021-02-18 163518](https://user-images.githubusercontent.com/59897162/108389481-a005bc00-7207-11eb-82c8-43b346be4387.jpg)

**Comment by 1aranzant** [comment]
> you need to link devices first

That's the issue, I get the same screen when trying to link a device.

**Comment by kinstray** [comment]
Yes, I'm getting those ridiculous pricing schemes too. My switch is still working with Octopi. Was it only a 1 month trial of the dev services? 

Got another e-mail you can use maybe?

**Comment by Senbei123** [comment]
Please just use the trial! You only need it for getting the local key. 
If you running home assistant as as example, you can run then the commands with a addon called local tuya without the cloud!

**Comment by vaderag** [comment]
Could someone please assist.

I run the access_token - all works great - success = true

I then paste {{url}}/v1.0/devices/{{device_id}} into the GET box... run it 
`{
    "code": 1010,
    "msg": "token invalid",
    "success": false,
    "t": 1614xxxxxxxxx
}`

What am I doing wrong?

Edit: nevermind. Seems there is a pre-set GET that has that in although i missed it due to all the chinese writing!!

**Comment by DEVAXTATOR** [comment]
please read devaxtator post

**Comment by matyman1122** [comment]
I am having trouble getting the API Explorer to work. I have linked device:
![image](https://user-images.githubusercontent.com/57550198/109965963-c701d900-7d43-11eb-9739-6a1b80372480.png)
I have tried the different regions as well China gets permission denied:
![image](https://user-images.githubusercontent.com/57550198/109966207-1516dc80-7d44-11eb-8431-644ccee94f3e.png)
Europe returns blank response
![image](https://user-images.githubusercontent.com/57550198/109966354-47c0d500-7d44-11eb-9f65-c4a8c9d5bca8.png)
My devices show in the device list under Europe selection
![image](https://user-images.githubusercontent.com/57550198/109966706-ac7c2f80-7d44-11eb-8d89-c65ca59ab6b6.png)

Anyone have ideas here? I need to create an account on the app under a different locations?

**Comment by DEVAXTATOR** [comment]
> I am having trouble getting the API Explorer to work. I have linked device:
> ![image](https://user-images.githubusercontent.com/57550198/109965963-c701d900-7d43-11eb-9739-6a1b80372480.png)
> I have tried the different regions as well China gets permission denied:
> ![image](https://user-images.githubusercontent.com/57550198/109966207-1516dc80-7d44-11eb-8431-644ccee94f3e.png)
> Europe returns blank response
> ![image](https://user-images.githubusercontent.com/57550198/109966354-47c0d500-7d44-11eb-9f65-c4a8c9d5bca8.png)
> My devices show in the device list under Europe selection
> ![image](https://user-images.githubusercontent.com/57550198/109966706-ac7c2f80-7d44-11eb-8d89-c65ca59ab6b6.png)
> 
> Anyone have ideas here? I need to create an account on the app under a different locations?

again....
go to api managment
api groups
and grant permissions to the api only DEVICE MANAGMENT wait 10 minutes
and then get device details is that easy
![Captura de ecrã 2021-03-04 125559](https://user-images.githubusercontent.com/59897162/109967247-1830af00-7ce9-11eb-9752-dc2d19c0d201.jpg)

**Comment by matyman1122** [comment]
Clearly from my screenshot, device management is enabled and open under the API Groups.
I think it is strange I am getting blank response from Europe region api?
Do you get blank response from any region?

**Comment by DEVAXTATOR** [comment]
confirmed the response is response{}
something is wrong

**Comment by matyman1122** [comment]
thanks for the confirmation, I will try again tomorrow

**Comment by kinstray** [comment]
If you are getting 'Permission Deny' read my post from the 5th Jan 2021...

https://github.com/codetheweb/tuyapi/issues/323#issuecomment-754492846

**Comment by matyman1122** [comment]
> If you are getting 'Permission Deny' read my post from the 5th Jan 2021...
> 
> [#323 (comment)](https://github.com/codetheweb/tuyapi/issues/323#issuecomment-754492846)

Thanks for the reply, this step is already completed as per my last message. You can see in my screenshot all the API groups are available for selection. I suspect I get the deny message because that deviceID is not available on the China API as my devices are under the Europe location. But when I use the Europe one I get {} response. I might need to raise a ticket to them to enquire the status of the API. I was not able to find a status page anywhere.

**Comment by matyman1122** [comment]
Just to let everyone know I decided to try and use postman to call the Europe API instead and this is working.
The issue is on the Tuya Cloud not having permissions to the API

**Comment by matyman1122** [comment]
For anyone that is trying to use the API Explorer through Tuya IOT Platform and getting the response {} I have found the issue.
This is because the connection is getting blocked due to CORS restrictions. I found this error in the developer console.
![image](https://user-images.githubusercontent.com/57550198/110327230-d055b300-806d-11eb-933c-d88d0adef868.png)

I was able to get the API working again using this extension for Chrome: 
https://chrome.google.com/webstore/detail/cross-domain-cors/mjhpgnbimicffchbodmgfnemoghjakai

**Reply by robertlacroix** [reply]
Great find, that fixed it for me! In Safari you can temporarily disable CORS restrictions in the Develop menu, but make sure to enable it again after you're done.

<img width="270" alt="Screen Shot 2021-03-14 at 9 22 38 PM" src="https://user-images.githubusercontent.com/550277/111103314-67f25000-850b-11eb-94f9-a693f3468d30.png">

**Comment by ground-creative** [comment]
Here is a simple lib to connect to the api and retrieve local keys of devices:

https://github.com/ifsale/tuyapiphp

**Comment by rayr007** [comment]
Calling "Get Device details" here worked for me:
<img width="265" alt="image" src="https://user-images.githubusercontent.com/6305569/137560843-822adc7e-0fa9-4fdd-97e5-bd9bdc01369e.png">

**Reply by rayr007** [reply]
While this one returns permission denied:
<img width="250" alt="image" src="https://user-images.githubusercontent.com/6305569/137561026-50c16e33-6108-4257-954d-f596203377b2.png">

**Reply by vtipalko** [reply]
General Device Capabilities has worked! Thank you!!

And do you use local tuya or the original one?

![Snímka obrazovky 2023-03-04 124602](https://user-images.githubusercontent.com/123496207/222898548-41868e8d-6fb4-4b99-9bb8-56652c6d49ee.png)
because now when the trail ended, the originial doesn't show any devices. 

the local one now shoes the devices as before but i cant control it. it is faded
![Snímka obrazovky 2023-03-04 124723](https://user-images.githubusercontent.com/123496207/222898544-e1b2b1d3-6ddc-42e5-b4a4-c2845d8462d7.png)

**Comment by vtipalko** [comment]
Do i need to pay to retrieve the API key? or am i just dumb : ((


![Snímka obrazovky 2023-02-26 220437](https://user-images.githubusercontent.com/123496207/221437229-2306176c-4557-4c7d-8282-cea7c7fa5324.jpg)

**Reply by rayr007** [reply]
Be sure you have the correct data center at the top selected. It still works for me for free.

Both of these get me the local_key attribute:

"General Device Capabilities" > "General Devices management" > "Get Device Information" (not "Get the device information")
"Smart Home Device System" > "Device Management" > "Get device details"
